aten/src/ATen/native/mps/operations/FusedAdamKernel.mm

```diff
@@ -66,4 +63,79 @@ void _fused_adam_kernel_mps_(at::TensorList params,
   }
 }
 
+// The following overload simply has a Tensor lr
+void _fused_adam_kernel_mps_(at::TensorList params,
+                             at::TensorList grads,
+                             at::TensorList exp_avgs,
+                             at::TensorList exp_avg_sqs,
+                             at::TensorList max_exp_avg_sqs,
+                             at::TensorList state_steps,
+                             const at::Tensor& lr,
+                             const double beta1,
+                             const double beta2,
+                             const double weight_decay,
+                             const double eps,
+                             const bool amsgrad,
+                             const bool maximize,
+                             const std::optional<at::Tensor>& grad_scale,
+                             const std::optional<at::Tensor>& found_inf) {
+  if (lr.is_cpu()) {
+    _fused_adam_kernel_mps_(params,
+                            grads,
+                            exp_avgs,
+                            exp_avg_sqs,
+                            max_exp_avg_sqs,
+                            state_steps,
+                            lr.item<double>(),
+                            beta1,
+                            beta2,
+                            weight_decay,
+                            eps,
+                            amsgrad,
+                            maximize,
+                            grad_scale,
+                            found_inf);
+    return;
```

Comment:
Nit, use modern C++

Per C++14, 6.6.3 "The return statement":
> A return statement with an expression of non-void type can be used only in functions returning a value [...] A > >return statement with an expression of type void can be used only in functions with a return type of cv void; the >expression is evaluated just before the function returns to its caller.
```suggestion
    return _fused_adam_kernel_mps_(params,
                            grads,
                            exp_avgs,
                            exp_avg_sqs,
                            max_exp_avg_sqs,
                            state_steps,
                            lr.item<double>(),
                            beta1,
                            beta2,
                            weight_decay,
                            eps,
                            amsgrad,
                            maximize,
                            grad_scale,
                            found_inf);
```

Reply:
This kinda works out of my expectation...done.